src/marqo/tensor_search/add_docs.py

```diff
@@ -192,4 +196,51 @@ def determine_document_dict_field_type(field_name: str, field_content, mappings:
         else:
             raise base_exceptions.InternalError(f"Invalid dict field type: '{mappings[field_name]['type']}' for field: '{field_name}' in mappings. Must be one of {[t.value for t in enums.MappingsObjectType]}")
     else:
-        return None
\ No newline at end of file
+        return None
+    
+
+
+def determine_text_prefix(request_level_prefix: str, index_settings: IndexSettings, prefix_type: str) -> str:
+    """
+    Determines the text prefix to be used for chunking text fields or search queries.
+    This prefix will be added before each text chunk or query to enhance processing accuracy.
+        
+    Logic:
+    1. Prioritize request-level prefix
+    2. If not provided, use settings based on prefix_type
+    3. If still not provided, use model_properties defined prefix
+    4. Return None if no prefix is found, handling is expected by the caller
+
+    Args:
+        request_level_prefix (str): The prefix provided in the request
+        index_settings (IndexSettings): The settings object containing prefix information
+        prefix_type (str): Either "text_query_prefix" or "text_chunk_prefix"
+
+    Returns:
+        str: The determined prefix, or None if no prefix is found
+    """
+
+    if request_level_prefix is not None:
+        return request_level_prefix
+
+    # Use the appropriate prefix from index_settings based on prefix_type
+    if prefix_type == "text_query_prefix" and index_settings.text_query_prefix is not None:
+        return index_settings.text_query_prefix
+    elif prefix_type == "text_chunk_prefix" and index_settings.text_chunk_prefix is not None:
+        return index_settings.text_chunk_prefix
+
+
+    # Fallback to model_properties defined prefix
+    model_properties = get_model_properties_from_registry(index_settings.model)
```

Comment:
2 things to note:

1. It looks like `MarqoIndex` doesn't have an `IndexSettings` property. Instead, the `Model`, `TextPreProcessing`, etc, are direct properties of it: https://github.com/marqo-ai/marqo/blob/6e40f6ccb5896d14229b86b54f59383bf1ce3c8b/src/marqo/core/models/marqo_index.py#L193
2. The `Model` object has its own attribute `properties` so you probably want to check that first: https://github.com/marqo-ai/marqo/blob/6e40f6ccb5896d14229b86b54f59383bf1ce3c8b/src/marqo/core/models/marqo_index.py#L121